report/topology.go

```diff
@@ -41,28 +41,28 @@ type EdgeMetadata struct {
 }
 
 // NodeMetadata describes a superset of the metadata that probes can collect
-// about a given node in a given topology. Right now it's a weakly-typed map,
-// which should probably change (see comment on type MapFunc).
-type NodeMetadata map[string]string
+// about a given node in a given topology.
+type NodeMetadata struct {
+	Metadata
+}
 
-// Copy returns a value copy, useful for tests.
-func (nm NodeMetadata) Copy() NodeMetadata {
-	cp := make(NodeMetadata, len(nm))
-	for k, v := range nm {
-		cp[k] = v
+// Metadata is shorthand for a string:string map.
+type Metadata map[string]string
```

Comment:
I originally did that but having map[string]string everywhere was messy.  Metadata is much easier to type.